src/test/java/com/iota/iri/service/milestone/impl/MilestoneServiceImplTest.java

```diff
@@ -0,0 +1,177 @@
+package com.iota.iri.service.milestone.impl;
+
+import com.iota.iri.TangleMockUtils;
+import com.iota.iri.controllers.MilestoneViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.HashFactory;
+import com.iota.iri.model.persistables.Transaction;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.snapshot.impl.SnapshotMockUtils;
+import com.iota.iri.storage.Tangle;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Optional;
+
+@RunWith(MockitoJUnitRunner.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class MilestoneServiceImplTest {
+    //region [CONSTANTS FOR THE TEST] //////////////////////////////////////////////////////////////////////////////////
+
+    private enum MockedMilestone {
+        A("ARWY9LWHXEWNL9DTN9IGMIMIVSBQUIEIDSFRYTCSXQARRTVEUFSBWFZRQOJUQNAGQLWHTFNVECELCOFYB", 70001),
+        B("BRWY9LWHXEWNL9DTN9IGMIMIVSBQUIEIDSFRYTCSXQARRTVEUFSBWFZRQOJUQNAGQLWHTFNVECELCOFYB", 70002),
+        C("CRWY9LWHXEWNL9DTN9IGMIMIVSBQUIEIDSFRYTCSXQARRTVEUFSBWFZRQOJUQNAGQLWHTFNVECELCOFYB", 70003),
+        D("JRWY9LWHXEWNL9DTN9IGMIMIVSBQUIEIDSFRYTCSXQARRTVEUFSBWFZRQOJUQNAGQLWHTFNVECELCOFYB", 70010),
+        E("KRWY9LWHXEWNL9DTN9IGMIMIVSBQUIEIDSFRYTCSXQARRTVEUFSBWFZRQOJUQNAGQLWHTFNVECELCOFYB", 70011),
+        F("LRWY9LWHXEWNL9DTN9IGMIMIVSBQUIEIDSFRYTCSXQARRTVEUFSBWFZRQOJUQNAGQLWHTFNVECELCOFYB", 70012);
+
+        private final Hash transactionHash;
+
+        private final int milestoneIndex;
+
+        MockedMilestone(String transactionHash, int milestoneIndex) {
+            this.transactionHash = HashFactory.TRANSACTION.create(transactionHash);
+            this.milestoneIndex = milestoneIndex;
+        }
+
+        public void mockProcessed(Tangle tangle, boolean applied) {
+            TangleMockUtils.mockMilestone(tangle, transactionHash, milestoneIndex);
+            Transaction mockedTransaction = TangleMockUtils.mockTransaction(tangle, transactionHash);
+            mockedTransaction.snapshot = applied ? milestoneIndex : 0;
+        }
+    }
+
+    //endregion ////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    //region [BOILERPLATE] /////////////////////////////////////////////////////////////////////////////////////////////
+
+    @Mock
+    private Tangle tangle;
+
+    @Mock
+    private SnapshotProvider snapshotProvider;
+
+    @InjectMocks
+    private MilestoneServiceImpl milestoneService;
+
+    @Before
+    public void setUp() {
+        SnapshotMockUtils.mockSnapshotProvider(snapshotProvider);
+
+        MilestoneViewModel.clear();
```

Comment:
Other tests don't clean up the cached MilestoneViewModels so it can happen that we start this test with some cache Milestones, but you are right this should ALSO go into after the test.